utils/joiner/tuplejoiner.cpp

```diff
@@ -516,38 +693,47 @@ void TupleJoiner::doneInserting()
 
         rowCount = size();
 
+        uint bucket = 0;
         if (joinAlg == PM)
             pmpos = 0;
         else if (typelessJoin)
-            thit = ht->begin();
+            thit = ht[bucket]->begin();
         else if (smallRG.getColType(smallKeyColumns[0]) == CalpontSystemCatalog::LONGDOUBLE)
-            ldit = ld->begin();
+            ldit = ld[bucket]->begin();
         else if (!smallRG.usesStringTable())
-            hit = h->begin();
+            hit = h[bucket]->begin();
         else
-            sthit = sth->begin();
+            sthit = sth[bucket]->begin();
 
         for (i = 0; i < rowCount; i++)
```

Comment:
I wonder if this loop is called often and for large amounts of Rows?

Reply:
Yeah this was one of the uglier mods i made for this.  At the end of small-table construction, it will iterate over all of the rows in the small table, stopping when it sees a certain # of unique values.  I can't say how often it scans the whole thing in real use cases.

Reply:
What is the value for this #?

Reply:
The total # of rows on the small side as returned by TupleJoiner::size().